src/components/Widgets/ControlHOC.js

```diff
@@ -80,14 +81,25 @@ class ControlHOC extends Component {
   };
 
   render() {
-    const { controlComponent, field, value, metadata, onChange, onAddAsset, onRemoveAsset, getAsset } = this.props;
+    const {
+      controlComponent,
+      field,
+      value,
+      metadata,
+      onChange,
+      onAddAsset,
+      onRemoveAsset,
+      getAsset,
+      registerDependency,
+    } = this.props;
     return React.createElement(controlComponent, {
       field,
       value,
       metadata,
       onChange,
       onAddAsset,
       onRemoveAsset,
+      registerDependency,
```

Comment:
seems like `onAddDependency` and `onRemoveDependency` would be more in line with the other callback functions in terms of naming (and it should be possible to remove a dependency as well as adding one...)